### PR TITLE
Initial PR for code review

### PR DESCRIPTION
**Pull request by untergeek**
My changes are here for review.  There are no tests yet.

### CODE REVIEW

File: logstash-core/lib/logstash/config/source/modules.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      module_names = []
+      modules_array.each {|module_hash| module_names << module_hash["name"]}
+      if module_names.length > module_names.uniq.length
+        raise LogStash::ConfigLoadingError, I18n.t("logstash.modules.configuration.modules-must-be-unique", :modules => module_names)
+      end
+      ### Here is where we can force the modules_array to use only [0] for 5.5, and leave
+      ### a warning/error message to that effect.
+      modules_array.each do |module_hash|
+        begin
+          thismodule = plugin_modules.find { |allmodules| allmodules.modulename == module_hash["name"] }
```
Comment (suyograo) [inline logstash-core/lib/logstash/config/source/modules.rb]: `thismodule` => `current_module`

File: logstash-core/lib/logstash/config/source/modules.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          altname = "module-#{module_hash["name"]}"
```
Comment (suyograo) [inline logstash-core/lib/logstash/config/source/modules.rb]: `altname` -> `alt_name` for ruby style.

File: logstash-core/lib/logstash/config/source/modules.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          pipeline_id = module_hash["pipeline_id"].nil? ? altname : module_hash["pipeline_id"]
```
Comment (suyograo) [inline logstash-core/lib/logstash/config/source/modules.rb]: We shouldn't even let users specify `pipeline_id`, to simplify -- for 6.0 as well as 5.x. We can infer from the module name as you've already done (`altname`). One less config to worry about.

File: logstash-core/lib/logstash/config/source/modules.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          config_string = thismodule.config_string(module_hash)
+          logger.debug("CONFIG_STRING", :config_string => config_string)
```
Comment (suyograo) [inline logstash-core/lib/logstash/config/source/modules.rb]: needs to be more descriptive. 

"Config string for module", :config_string, :module => name"

File: logstash-core/lib/logstash/modules_cli_parser.rb [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+  def decommafy(value)
+    retval = []
+    if value.split(',').length == 1
+      retval << value
+    elsif value.split(',').length > 1
+      value.split(',').each {|subvalue| retval << subvalue}
+    end
+    retval
+  end
```
Comment (ph) [inline logstash-core/lib/logstash/modules_cli_parser.rb]: I don't see the unit test for this method, but I think You don't need to have a conditional here and just return `value.split(',')`

File: logstash-core/lib/logstash/modules_cli_parser.rb [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    if unparsed.split('=').length >= 2
+      # This hackery is to catch the possibility of an equals (`=`) sign 
+      # in a passphrase, which might result in an incomplete key.  The 
+      # portion before the first `=` should always be the key, leaving 
+      # the rest to be the value
+      values = unparsed.split('=')
+      k = values.shift
+      # v = values.join('=')
+    else
+      raise LogStash::ConfigLoadingError, I18n.t("logstash.modules.configuration.modules-variables-malformed", :rawvar => (module_name + '.' + unparsed))
```
Comment (ph) [inline logstash-core/lib/logstash/modules_cli_parser.rb]: thanks for all the feedback for the user!

File: logstash-core/lib/logstash/modules_cli_parser.rb [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    end
+    return k,values.join('=')
```
Comment (ph) [inline logstash-core/lib/logstash/modules_cli_parser.rb]: Can we move the return into the if statement so we get out early.

File: logstash-core/lib/logstash/runner.rb [hunk 8]
```diff
@@ -248,6 +261,10 @@ def execute
 
     return start_shell(setting("interactive"), binding) if setting("interactive")
 
+    module_parser = LogStash::ModulesCLIParser.new(@modules_list, @modules_variable_list)
+    # Now populate Setting for modules.list with our parsed array.
+    @settings.set("modules.config", module_parser.output)
+
```
Comment (ph) [inline logstash-core/lib/logstash/runner.rb]: Nice, I would have expected it to be more hackery but this is a nice way to do it.

File: logstash-core/lib/logstash/environment.rb [hunk 9]
```diff
@@ -20,6 +20,8 @@ module Environment
     Setting::NullableString.new("path.config", nil, false),
  Setting::WritableDirectory.new("path.data", ::File.join(LogStash::Environment::LOGSTASH_HOME, "data")),
     Setting::NullableString.new("config.string", nil, false),
+                    Setting.new("modules.config", Array, []),
```
Comment (untergeek) [inline logstash-core/lib/logstash/environment.rb]: Does `modules.name` make more sense? Or `modules.cli`?  I'm partial to `modules.cli`

Comment (suyograo) [inline logstash-core/lib/logstash/environment.rb]: I like `modules.cli`. Lets go with that.

File: logstash-core/lib/logstash/modules.rb [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    def setting(value, default)
+      @settings[value].nil? ? default : @settings[value]
+    end
+
+    def render
+      # process the template and settings
+      # send back as a string with no newlines (the '>' part)
+      renderer = ERB.new(File.read(@template), 3, '>')
+      renderer.result(binding)
+    end
```
Comment (untergeek) [inline logstash-core/lib/logstash/modules.rb]: Yep.

File: logstash-core/lib/logstash/modules_cli_parser.rb (hunk 5 above)
Comment (untergeek) [inline logstash-core/lib/logstash/modules_cli_parser.rb]: You're right.  Breaking this down into a class removed its need altogether.  I'll be resolving this with `#concat` and a single `split(',')` instead.

**Pull request closed without merging**